torch/csrc/autograd/variable.cpp

```diff
@@ -342,23 +353,15 @@ const c10::VariableVersion& version_counter(const Variable& self) {
 
 void add_hook(
     const at::TensorBase& self,
-    std::shared_ptr<FunctionPreHook> hook) {
-  materialize_autograd_meta(self)->hooks_.push_back(std::move(hook));
+    std::unique_ptr<FunctionPreHook> hook) {
+  AutogradMeta* meta = materialize_autograd_meta(self);
+  TORCH_INTERNAL_ASSERT(meta->hooks_.size() == 0);
```

Comment:
Why? You should be able to add multiple hooks no?

Reply:
Yes, but there shouldn't be a reason to. You shouldn't really be working with the hooks_ field directly. You should either be registering cpp tensor hooks or python tensor hooks, and those mechanisms should guarantee that there is only one hook in hook_ at a time.